libres/lib/enkf/enkf_main.cpp

```diff
@@ -743,6 +737,349 @@ static FILE *enkf_main_log_step_list(const char *log_path,
     return log_stream;
 }
 
+static std::map<std::string, matrix_type *>
+enkf_main_load_parameters_from_ministep(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int step2, run_mode_type run_mode,
+    meas_data_type *forecast, enkf_state_type **ensemble, hash_type *use_count,
+    obs_data_type *obs_data, const local_ministep_type *ministep) {
+
+    std::map<std::string, matrix_type *> parameters;
```

Comment:
Style: place this line right before usage rather than hiding it at the top C-style. Right above the `while` is probably the location for it.

The reason C people put all of their declarations at the top is historical. In _very_ early versions of C, you had to first specify all the variables that would be used in the function. This hasn't been a requirement since at least the first ISO C (early 80s), but has lived on somehow.

In C++ we prefer to put the declaration as close to the usage site as possible, as you would do in Python.